client/src/components/SigninButton.svelte

```diff
@@ -1,19 +1,151 @@
 <script lang="ts">
   import Modal from "./Modal.svelte";
   import { authStore as user } from "$stores/auth";
+  import { loginWithGoogle, loginWithMicrosoft, loginWithEmail, onSignOut, createUser } from "$lib/firebase/auth";
 
   // check if the user is logged in with getAuth
   let openSignInModal = false;
+  let email = "";
+  let password = "";
+  let confirmPass = "";
+  let clickedButton = false;
+  let clickedButton2 = false;
+  let errorMessage = "";
+  async function createAccount() {
+    if (password == "") {
+      return;
+    }
+    if (password != confirmPass) {
+      errorMessage = "Password does not match";
+      password = "";
+      confirmPass = "";
+      return;
+    }
+    if (email == "") {
+      return;
+    } else {
+      try {
+        await createUser(email, password);
+        errorMessage = "";
+      } catch (err) {
+        errorMessage = "Invalid Username or Password";
+      }
+    }
+
+    if (errorMessage == "") {
+      openSignInModal = false;
+    }
+    email = "";
+    password = "";
+    confirmPass = "";
+  }
+  async function submitLogin() {
+    if (password == "") {
+      return;
+    }
+    if (email == "") {
+      return;
+    } else
+      try {
+        await loginWithEmail(email, password);
+        errorMessage = "";
+      } catch (err) {
+        errorMessage = "Invalid Username or Password";
+      }
+    if (errorMessage == "") {
+      openSignInModal = false;
+    }
+    password = "";
+    email = "";
+  }
+  function signInWithEmailAndPassword() {
+    clickedButton = !clickedButton;
+    clickedButton2 = false;
+  }
+  function createWithEmailAndPassword() {
+    clickedButton2 = !clickedButton2;
+    clickedButton = false;
+  }
```

Comment:
it might be better to give these functions names that more closly reflect there functionallity. they are toggeling the visiblity of forms not really signing in or creating an account

Reply:
resolved